src/livekit/useLiveKit.ts

```diff
@@ -102,27 +125,32 @@ export function useLiveKit(): LiveKitState | undefined {
   return state;
 }
 
+// if a room is passed this only affects the device selection inside a call. Without room it changes what we see in the lobby
 function useMediaDevicesState(room: Room): MediaDevicesState {
+  let connectedRoom: Room;
+  if (room.state !== ConnectionState.Disconnected) {
+    connectedRoom = room;
+  }
```

Comment:
What if we move it to the `useLiveKitState()` function above and accept a `room?: Room`? I think it will be much more flexible and elegant since we'll then be able to use the function from multiple places (also useful for the further refactoring).